stdlib/postgres/postgres.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"net"
+
+	"cloud.google.com/go/cloudsqlconn"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/stdlib"
+)
+
+func init() {
+	sql.Register("cloudsql-postgres", &pgDriver{})
+}
+
+type pgDriver struct{}
+
+func (*pgDriver) Open(name string) (driver.Conn, error) {
+	config, err := pgx.ParseConfig(name)
+	if err != nil {
+		return nil, err
+	}
+	h := config.Config.Host          // Extract instance connection name
+	config.Config.Host = "localhost" // Replace it with a default value
+	config.DialFunc = func(ctx context.Context, _, _ string) (net.Conn, error) {
+		return cloudsqlconn.Dial(ctx, h)
```

Comment:
Does the user have any ability to customize the Dial connection here? What's the user journey if they want to use private IP? or IAM AuthN?

Reply:
It would be possible to add support for additional parameters in the DSN using the `RuntimeParams` in the [`Config`](https://pkg.go.dev/github.com/jackc/pgconn?utm_source=godoc#Config) struct, which works easily with [`ParseConfig`](https://pkg.go.dev/github.com/jackc/pgconn?utm_source=godoc#ParseConfig).

However, I think we should encourage users to configure the dialer directly and take feature requests for tuning this driver on a case by case basis.